### ISSUE TITLE
Make it possible to sort by datatype

### ISSUE THREAD

**Issue by frikky** [fhightower/ioc-finder#218]
Hey! I'm a part of the team running [Shuffle Automation](https://github.com/shuffle/shuffle), and we've been using this amazing library for a while. [Exact code location here](https://github.com/Shuffle/python-apps/blob/7589eb1823a2df172eee8859b49fe799b5587235/shuffle-tools/1.2.0/src/app.py#L240).

We've recently hit upon an optimization issue though, which happens at megabytes of data parsed straight in. So I was wondering; how much work would it be to add a way to find specific IOCs? E.g:
```
find_iocs("some email@email.com", data_types=["email_addresses", "domains"])
```

I had a look at the code and wanted to add it myself but weren't able to build it easily, so thought to ask you first (haven't run Python with virtualenv and such in a while). 

PS: I also realized you've built a lot since your last Pypi release (6.0.1) - any idea when you'll release, or how we could easily build the current release(s) into our Python pipelines?

Thanks :)

**Comment by frikky** [comment]
> Hi @frikky - Thanks for reaching out and thank you so much for the sponsorship heart tada ! That is really encouraging and motivating. Also, glad to hear this library is useful (and Shuffle looks very cool, I'll have to look into it a bit more).
> 
> ### RE: Support Parsing Specific Data Types
> As to your request about adding support for parsing specific data types, I have two thoughts:
> 
> 1. This is definitely possible and would be relatively easy to implement
> 2. Based on [the original issue](https://github.com/Shuffle/Shuffle/issues/865), the root problem is the slowness of this library. My next focus, once the newest version is released, is to [speed up this library](https://github.com/fhightower/ioc-finder/issues/196) as it's much slower than I want it to be
> 
> Both of these seem like good features to implement b/c they allow the library to run faster (albeit in different ways). These will be a priority for subsequent work (once the next version is released - see more in the next section).
> 
> ### RE: Releasing a new version
> As to releasing the next version, it should be released soon. I apologize for the delay; I usually try to keep releases pretty quick, but this one has dragged on as you noted. I've started tracking the remaining work to be done before releasing version `7.0.0` [here](https://github.com/users/fhightower/projects/2/views/1?layout=board) (there's just one more issue which I hope to resolve soon).
> 
> Does that help? Thanks again for raising this issue as it touches on very relevant issues.

Heyo! This initial sponsorship will hopefully help you get some more - that's how it worked for me a few years back ;)

What would be required to getting the feature merged if I fix it? I agree it looks very simple (some more args + a switch/else if's), and would very much want it to be part of the next release so we can use it asap. Tests maybe?

Looking forward to the next version either way! We'll be using and breaking it :+1:

**Comment by fhightower** [comment]
Merging this feature would be pretty straight forward. Other than the implementation, the only things I'm looking are:

1. Tests validating common and edge cases
2. Updated documentation noting this argument (this can be ticketed for later)

And I recently released version `7.0.0` 🎉 , so feel free to beat it up and let me know how I can improve it 😄

**Comment by fhightower** [comment]
Fixed w/ #219 .

**Comment by fhightower** [comment]
@frikky : Want to follow up on this to note that when I added this feature, I introduced a bug that caused significant unnecessary slowness (very sorry!). I just released version `7.2.1` to fix this - so please upgrade to `7.2.1` ASAP. Thanks and sorry about the inconvenience.
